### DISCUSSION TITLE
cross-database populate not working

### DISCUSSION THREAD

**Discussion by ayalhus** [Automattic/mongoose#9949] [Q&A] [answered]
Trying to populate postDB collection with user's information from userDB: 

`mongoose.connect("mongodb://localhost:27017/userDB", { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.set("useUnifiedTopology", true);
 mongoose.set("useCreateIndex", true);

 const postDbConnection = mongoose.createConnection('mongodb://localhost:27017/postDB', { useNewUrlParser: true });

const userSchema = new mongoose.Schema({

  email: String,
  password: String,
  googleId: String,
 posts:  [{ type: mongoose.Schema.Types.ObjectId, ref: 'Post' }]
});

const postSchema = new mongoose.Schema({
  title: String,
  content: String,
  created:
  {type: Date, default: Date.now},
  user: [{ type: mongoose.Schema.Types.ObjectId, ref: User, required: true }]
});

userSchema.plugin(passportLocalMongoose);
userSchema.plugin(findOrCreate);

// this uses the default connection (to `userDB`)
const User = new mongoose.model("User", userSchema);

// this uses the second connection (to `postDb`)
const Post = postDbConnection.model('Post', postSchema);`

I get the following error:

Cannot access 'User' before initialization

Unless I use
  ref: 'User'
instead of
    ref: User
 Which is not the correct way to refer to a model name from an outside connection

**Comment by vkarpov15** [comment, marked as answer]
Use 'User'. Both ways are correct depending on context, but in this case you can't use `ref: User` because `User` isn't defined when you're trying to use it.
